File: lib/homepage.dart
import 'package:flutter/material.dart';
import 'header.dart';
import 'footer.dart';

const Color charcoal = Color(0xFF333333);
const Color oliveGreen = Color(0xFF97B469);
const Color creamWhite = Color(0xFFF8F5F0);

class HomePage extends StatefulWidget {
  const HomePage({Key? key}) : super(key: key);
  static const String routeName = '/home';

  @override
  HomePageState createState() => HomePageState();
}

class HomePageState extends State<HomePage> {
  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
  final Set<String> _hoveredTiles = {};

  // Screen size helpers
  bool get isMobile => MediaQuery.of(context).size.width < 600;
  bool get isTablet => MediaQuery.of(context).size.width >= 600 && MediaQuery.of(context).size.width < 900;
  bool get isDesktop => MediaQuery.of(context).size.width >= 900;
  bool get isVerySmallMobile => MediaQuery.of(context).size.width <= 350;

  @override
  Widget build(BuildContext context) {
    return BaseScaffold(
      scaffoldKey: _scaffoldKey,
      showBottomBorder: false,
      background: creamWhite,
      extendBehind: true,
      body: Stack(
        children: [
          Container(color: creamWhite),
          SingleChildScrollView(
            physics: const BouncingScrollPhysics(),
            child: Column(
              children: [
                _buildBannerSection(context),
                const SizedBox(height: 8),
                _buildContentSection(context),
                const SizedBox(height: 8),
                _buildFilterSection(context),
                _buildTileSection(context),
                const Footer(),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildBannerSection(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(isVerySmallMobile ? 12 : isMobile ? 16 : 20),
      width: double.infinity,
      height: MediaQuery.of(context).size.height * (isVerySmallMobile ? 0.3 : isMobile ? 0.32 : 0.35),
      decoration: const BoxDecoration(
        image: DecorationImage(
          image: AssetImage("background.png"),
          fit: BoxFit.cover,
        ),
      ),
      child: Column(
        children: [
          const Spacer(),
          Container(
            width: MediaQuery.of(context).size.width * (isVerySmallMobile ? 0.9 : isMobile ? 0.85 : 0.7),
            constraints: const BoxConstraints(maxWidth: 500),
            decoration: BoxDecoration(
              color: creamWhite,
              borderRadius: BorderRadius.circular(isVerySmallMobile ? 30 : isMobile ? 35 : 40),
              boxShadow: const [
                BoxShadow(
                  color: Colors.black26,
                  blurRadius: 4,
                  offset: Offset(0, 2),
                ),
              ],
            ),
            padding: const EdgeInsets.symmetric(horizontal: 10),
            child: Row(
              children: [
                Icon(Icons.search, color: charcoal, size: isVerySmallMobile ? 18 : isMobile ? 20 : 22),
                Expanded(
                  child: TextField(
                    textAlign: TextAlign.center,
                    style: TextStyle(
                      fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                    ),
                    decoration: InputDecoration(
                      hintText: 'What do you want to read?',
                      hintStyle: TextStyle(
                        fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                      ),
                      border: InputBorder.none,
                      contentPadding: EdgeInsets.symmetric(vertical: isVerySmallMobile ? 8 : 12),
                    ),
                  ),
                ),
                Icon(Icons.mic, color: charcoal, size: isVerySmallMobile ? 18 : isMobile ? 20 : 22),
              ],
            ),
          ),
        ],
      ),
    );
  }

 Widget _buildContentSection(BuildContext context) {
    return Container(
      margin: EdgeInsets.symmetric(
        horizontal: isVerySmallMobile ? 12 : isMobile ? 16 : 24,
        vertical: 8,
      ),
      padding: EdgeInsets.all(isVerySmallMobile ? 10 : isMobile ? 12 : 16),
      width: MediaQuery.of(context).size.width * (isVerySmallMobile ? 0.95 : isMobile ? 0.92 : 0.9),
      constraints: const BoxConstraints(maxWidth: 900),
      decoration: BoxDecoration(
        color: const Color(0x8297B469),
        borderRadius: BorderRadius.circular(isVerySmallMobile ? 12 : isMobile ? 14 : 16),
      ),
      child: LayoutBuilder(
        builder: (context, constraints) {
          // Determine grid columns based on screen size
          int crossAxisCount;
          double childAspectRatio;
          
          if (isVerySmallMobile) {
            crossAxisCount = 2;
            childAspectRatio = 1.1;
          } else if (isMobile) {
            crossAxisCount = 2;
            childAspectRatio = 1.2;
          } else if (isTablet) {
            crossAxisCount = 4; // 4 blocks in row for tablet
            childAspectRatio = 1.0; // Adjusted ratio to fit content
          } else {
            crossAxisCount = 4;
            childAspectRatio = 1.3;
          }

          return GridView.builder(
            padding: EdgeInsets.zero,
            shrinkWrap: true,
            physics: const NeverScrollableScrollPhysics(),
            gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
              crossAxisCount: crossAxisCount,
              crossAxisSpacing: isVerySmallMobile ? 8 : isMobile ? 12 : 16,
              mainAxisSpacing: isVerySmallMobile ? 8 : isMobile ? 12 : 16,
              childAspectRatio: childAspectRatio,
            ),
            itemCount: 4,
            itemBuilder: (context, index) {
              return _buildFeatureBox(
                title: [
                  "Continue where you left off",
                  "Correct Mistakes",
                  "Progress Rate",
                  "Streaks",
                ][index],
                content: [
                  _buildResumeButton(),
                  _buildProgressBar(),
                  _buildProgressText(),
                  _buildStreaksText(),
                ][index],
                alignLeft: index == 0,
              );
            },
          );
        },
      ),
    );
  }

  Widget _buildFilterSection(BuildContext context) {
    return Container(
      width: MediaQuery.of(context).size.width * (isVerySmallMobile ? 0.95 : isMobile ? 0.92 : 0.88),
      constraints: const BoxConstraints(maxWidth: 900),
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          ToggleButtons(
            isSelected: [true, false],
            selectedColor: creamWhite,
            color: Colors.black,
            fillColor: oliveGreen,
            onPressed: (index) {},
            constraints: BoxConstraints(
              minWidth: isVerySmallMobile ? 60 : isMobile ? 70 : 80,
              minHeight: isVerySmallMobile ? 32 : isMobile ? 34 : 36,
            ),
            children: [
              Padding(
                padding: EdgeInsets.symmetric(
                  horizontal: isVerySmallMobile ? 6 : isMobile ? 8 : 10,
                  vertical: 4,
                ),
                child: Text(
                  "Surah",
                  style: TextStyle(
                    fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                  ),
                ),
              ),
              Padding(
                padding: EdgeInsets.symmetric(
                  horizontal: isVerySmallMobile ? 6 : isMobile ? 8 : 10,
                  vertical: 4,
                ),
                child: Text(
                  "Juz",
                  style: TextStyle(
                    fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                  ),
                ),
              ),
            ],
          ),
          Row(
            children: [
              Text(
                "Sort by: ",
                style: TextStyle(
                  fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                ),
              ),
              DropdownButton<String>(
                value: "ASCENDING",
                underline: const SizedBox(),
                onChanged: (value) {},
                items: ["ASCENDING", "DESCENDING"].map((e) => DropdownMenuItem(
                  value: e,
                  child: Text(
                    e,
                    style: TextStyle(
                      fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                    ),
                  ),
                )).toList(),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildTileSection(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: 20),
      padding: EdgeInsets.all(isVerySmallMobile ? 8 : isMobile ? 10 : 12),
      width: MediaQuery.of(context).size.width * (isVerySmallMobile ? 0.95 : isMobile ? 0.92 : 0.88),
      constraints: const BoxConstraints(maxWidth: 900),
      decoration: BoxDecoration(
        color: creamWhite,
        border: Border.all(color: oliveGreen, width: isTablet ? 1.25 : 1.5),
        borderRadius: BorderRadius.circular(isVerySmallMobile ? 10 : isMobile ? 12 : 14), // Reduced radius for tablet
      ),
      child: LayoutBuilder(
        builder: (context, constraints) {
          int crossAxisCount;
          if (constraints.maxWidth < 500) {
            crossAxisCount = 2;
          } else if (isTablet) {
            crossAxisCount = 3; // Changed to 3 for tablet
          } else {
            crossAxisCount = 3;
          }

          return GridView.count(
            padding: EdgeInsets.zero,
            shrinkWrap: true,
            physics: const NeverScrollableScrollPhysics(),
            crossAxisCount: crossAxisCount,
            crossAxisSpacing: isVerySmallMobile ? 6 : isMobile ? 8 : 10,
            mainAxisSpacing: isVerySmallMobile ? 6 : isMobile ? 8 : 10,
            childAspectRatio: isTablet ? 2.8 : isVerySmallMobile ? 2.5 : 3,
            children: [
              _buildTile("Al-Fatihah", "(The Opening)"),
              _buildTile("Al-Baqarah", "(The Cow)"),
              _buildTile("Al-Imran", "(The Family of Imran)"),
              _buildTile("An-Nisa", "(The Women)"),
              _buildTile("Al-Maidah", "(The Table Spread)"),
              _buildTile("Al-An'am", "(The Cattle)"),
              _buildTile("Al-A'raf", "(The Heights)"),
              _buildTile("Al-Anfal", "(The Spoils of War)"),
              _buildTile("At-Tawbah", "(The Repentance)"),
              _buildTile("Yunus", "(Jonah)"),
              _buildTile("Hud", "(Hud)"),
              _buildTile("Yusuf", "(Joseph)"),
            ],
          );
        },
      ),
    );
  }

  Widget _buildTile(String title, String subtitle) {
    return MouseRegion(
      onEnter: (_) => setState(() => _hoveredTiles.add(title)),
      onExit: (_) => setState(() => _hoveredTiles.remove(title)),
      child: InkWell(
        onTap: () => debugPrint("Tile Clicked: $title"),
        borderRadius: BorderRadius.circular(isVerySmallMobile ? 8 : isMobile ? 10 : 12), // Reduced radius
        child: AnimatedContainer(
          duration: const Duration(milliseconds: 200),
          decoration: BoxDecoration(
            color: _hoveredTiles.contains(title) ? oliveGreen.withOpacity(0.12) : creamWhite,
            border: Border.all(color: oliveGreen, width: isTablet ? 1.25 : 1.5),
            borderRadius: BorderRadius.circular(isVerySmallMobile ? 8 : isMobile ? 10 : 12), // Reduced radius
          ),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Text(
                title,
                style: TextStyle(
                  fontWeight: FontWeight.bold,
                  fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16,
                ),
                textAlign: TextAlign.center,
                maxLines: 1,
                overflow: TextOverflow.ellipsis,
              ),
              const SizedBox(height: 4),
              Text(
                subtitle,
                style: TextStyle(
                  fontSize: isVerySmallMobile ? 10 : isMobile ? 12 : 14,
                ),
                textAlign: TextAlign.center,
                maxLines: 1,
                overflow: TextOverflow.ellipsis,
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildFeatureBox({
    required String title,
    required Widget content,
    bool alignLeft = false,
  }) {
    return Container(
      padding: EdgeInsets.all(isVerySmallMobile ? 8 : isMobile ? 10 : isTablet ? 10 : 12),
      decoration: BoxDecoration(
        color: creamWhite,
        borderRadius: BorderRadius.circular(isVerySmallMobile ? 10 : isMobile ? 12 : isTablet ? 12 : 14),
        boxShadow: const [
          BoxShadow(color: Colors.black26, blurRadius: 4, offset: Offset(0, 2)),
        ],
      ),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: alignLeft ? CrossAxisAlignment.start : CrossAxisAlignment.center,
        children: [
          Text(
            title,
            textAlign: alignLeft ? TextAlign.left : TextAlign.center,
            style: TextStyle(
              fontWeight: FontWeight.bold,
              fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : isTablet ? 14 : 16,
            ),
            maxLines: 2,
            overflow: TextOverflow.ellipsis,
          ),
          const Spacer(),
          // Special handling for tablet content sizing
          if (isTablet) 
            SizedBox(
              height: 60, // Fixed height for tablet content
              child: content,
            )
          else
            content,
        ],
      ),
    );
  }

  Widget _buildResumeButton() {
    return Align(
      alignment: Alignment.centerLeft,
      child: ElevatedButton(
        onPressed: () {},
        style: ElevatedButton.styleFrom(
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(isVerySmallMobile ? 18 : isMobile ? 20 : 22), // Reduced radius
          ),
          backgroundColor: oliveGreen,
          padding: EdgeInsets.symmetric(
            vertical: isVerySmallMobile ? 4 : isMobile ? 6 : 8,
            horizontal: isVerySmallMobile ? 10 : isMobile ? 12 : 16,
          ),
        ),
        child: Text(
          "Resume",
          style: TextStyle(
            color: creamWhite,
            fontSize: isVerySmallMobile ? 10 : isMobile ? 12 : 14,
          ),
        ),
      ),
    );
  }

  Widget _buildProgressBar() {
    return Column(
      children: [
        LinearProgressIndicator(
          value: 4/24,
          color: oliveGreen,
          backgroundColor: Colors.grey.shade300,
        ),
        const SizedBox(height: 6),
        Text(
          "20/24",
          style: TextStyle(fontSize: isVerySmallMobile ? 12 : isMobile ? 14 : 16),
        ),
      ],
    );
  }

  Widget _buildProgressText() => Text(
    "%",
    style: TextStyle(
      color: oliveGreen,
      fontSize: isVerySmallMobile ? 20 : isMobile ? 22 : 24, // Reduced size for tablet
    ),
  );

  Widget _buildStreaksText() => Text(
    "Days",
    style: TextStyle(
      color: oliveGreen,
      fontSize: isVerySmallMobile ? 20 : isMobile ? 22 : 24, // Reduced size for tablet
    ),
  );
}

File: src/pages/homepage.js
import React, { useState, useEffect } from 'react';
import Header from '../components/header';
import Footer from '../components/footer';
import './homepage.css';

const HomePage = () => {
  const [, setHoveredTiles] = useState(new Set()); // Removed unused hoveredTiles
  const [activeTab, setActiveTab] = useState('Surah');
  const [windowSize, setWindowSize] = useState({
    width: window.innerWidth,
    height: window.innerHeight,
  });
  
  useEffect(() => {
    document.body.classList.add('homepage');
    return () => {
      document.body.classList.remove('homepage');
      document.body.style.overflowY = ''; // Reset on unmount
    };
  }, []);

  useEffect(() => {
    const handleResize = () => {
      setWindowSize({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    };



    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  // Screen size helpers
  const isVerySmallMobile = windowSize.width <= 350;
  const isMobile = windowSize.width < 600;
  const isTablet = windowSize.width >= 600 && windowSize.width < 900;
  // Removed unused isDesktop variable

  const handleTileHover = (tile, isHovering) => {
    setHoveredTiles(prev => {
      const newSet = new Set(prev);
      isHovering ? newSet.add(tile) : newSet.delete(tile);
      return newSet;
    });
  };

  return (
    <div className="home-page">
      {/* Header is now absolutely positioned over the banner */}
      <Header transparent />
      
      {/* Banner Section - starts from top */}
      <div 
        className="banner-section" 
        style={{ 
          height: isVerySmallMobile ? '30vh' : isMobile ? '32vh' : '35vh',
          padding: isVerySmallMobile ? '12px' : isMobile ? '16px' : '20px'
        }}
      >
        <div 
          className="search-container"
          style={{
            width: isVerySmallMobile ? '90%' : isMobile ? '85%' : '70%',
            maxWidth: '500px',
            borderRadius: isVerySmallMobile ? '30px' : isMobile ? '35px' : '40px'
          }}
        >
          <span className="material-icons" style={{ 
            color: '#333333',
            fontSize: isVerySmallMobile ? '18px' : isMobile ? '20px' : '22px'
          }}>search</span>
          <input 
            type="text" 
            placeholder="What do you want to read?"
            style={{
              fontSize: isVerySmallMobile ? '12px' : isMobile ? '14px' : '16px',
              padding: isVerySmallMobile ? '8px 0' : '12px 0'
            }}
          />
          <span className="material-icons" style={{ 
            color: '#333333',
            fontSize: isVerySmallMobile ? '18px' : isMobile ? '20px' : '22px'
          }}>mic</span>
        </div>
      </div>

      <div className="content-container">
        {/* Content Section */}
        <div className="content-section"
  style={{
    margin: isVerySmallMobile ? '12px 16px' : isMobile ? '16px 20px' : '24px auto',
    padding: isVerySmallMobile ? '10px' : isMobile ? '12px' : '16px',
    width: 'calc(100% - 32px)',
    maxWidth: '900px',
    borderRadius: isVerySmallMobile ? '12px' : isMobile ? '14px' : '16px'
  }}
>
  <div className="grid-container">
    {[
      { title: "Continue where you left off", content: "resume", align: "left" },
      { title: "Correct Mistakes", content: "progress", align: "center" },
      { title: "Progress Rate", content: "percentage", align: "center" },
      { title: "Streaks", content: "days", align: "center" }
    ].map((item, index) => (
      <div 
        key={index}
        className="feature-box"
        style={{
          padding: isVerySmallMobile ? '8px' : isMobile ? '10px' : isTablet ? '10px' : '12px',
          borderRadius: isVerySmallMobile ? '10px' : isMobile ? '12px' : isTablet ? '12px' : '14px',
        }}
      >
        <h3 style={{
          fontSize: isVerySmallMobile ? '12px' : isMobile ? '14px' : isTablet ? '14px' : '16px',
          textAlign: item.align,
          marginBottom: '12px'
        }}>{item.title}</h3>
        
        <div className="feature-content" style={{ 
          height: isTablet ? '60px' : 'auto',
          width: '100%',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'flex-end'
        }}>
          {item.content === "resume" && (
            <button 
              className="resume-button"
              style={{
                borderRadius: isVerySmallMobile ? '18px' : isMobile ? '20px' : '22px',
                padding: isVerySmallMobile ? '4px 10px' : isMobile ? '6px 12px' : '8px 16px',
                fontSize: isVerySmallMobile ? '10px' : isMobile ? '12px' : '14px',
                alignSelf: 'flex-start'
              }}
            >
              Resume
            </button>
          )}
          {item.content === "progress" && (
            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
              <div className="mistakes-bar-container" style={{ width: '100%', maxWidth: '200px' }}>
                <div className="mistakes-bar">
                  <div className="mistakes-progress" style={{ width: '80%' }}></div>
                </div>
                <div className="mistakes-count" style={{ 
                  fontSize: isVerySmallMobile ? '10px' : isMobile ? '11px' : '12px',
                  marginTop: '4px',
                  textAlign: 'center'
                }}>24/30</div>
              </div>
            </div>
          )}
          {(item.content === "percentage" || item.content === "days") && (
            <p style={{
              color: '#97B469',
              fontSize: isVerySmallMobile ? '20px' : isMobile ? '22px' : '24px',
              textAlign: 'center',
              margin: '0 auto'
            }}>{item.content === "percentage" ? "%" : "Days"}</p>
          )}
        </div>
      </div>
    ))}
  </div>
</div>

        {/* Filter Section */}
        <div 
          className="filter-section"
          style={{
            width: isVerySmallMobile ? '95%' : isMobile ? '92%' : '88%',
            maxWidth: '900px',
            marginBottom: '16px'
          }}
        >
          <div className="tab-container">
            <div 
              className={`tab ${activeTab === 'Surah' ? 'active' : ''}`}
              onClick={() => setActiveTab('Surah')}
            >
              Surah
            </div>
            <div 
              className={`tab ${activeTab === 'Juz' ? 'active' : ''}`}
              onClick={() => setActiveTab('Juz')}
            >
              Juz
            </div>
          </div>
          <div className="sort-dropdown">
            <span style={{ fontSize: isVerySmallMobile ? '12px' : isMobile ? '14px' : '16px' }}>Sort by: </span>
            <select style={{ fontSize: isVerySmallMobile ? '12px' : isMobile ? '14px' : '16px' }}>
              <option>ASCENDING</option>
              <option>DESCENDING</option>
            </select>
          </div>
        </div>

        {/* Tile Section */}
        <div 
          className="tile-section"
          style={{
            padding: isVerySmallMobile ? '8px' : isMobile ? '10px' : '12px',
            width: isVerySmallMobile ? '95%' : isMobile ? '92%' : '88%',
            maxWidth: '900px',
            borderRadius: isVerySmallMobile ? '10px' : isMobile ? '12px' : '14px',
            borderWidth: isTablet ? '1.25px' : '1.5px',
            marginBottom: '40px' /* Added space before footer */
          }}
        >
          <div className="tile-grid">
            {[
              { title: "Al-Fatihah", subtitle: "(The Opening)" },
              { title: "Al-Baqarah", subtitle: "(The Cow)" },
              { title: "Al-Imran", subtitle: "(The Family of Imran)" },
              { title: "An-Nisa", subtitle: "(The Women)" },
              { title: "Al-Maidah", subtitle: "(The Table Spread)" },
              { title: "Al-An'am", subtitle: "(The Cattle)" },
              { title: "Al-A'raf", subtitle: "(The Heights)" },
              { title: "Al-Anfal", subtitle: "(The Spoils of War)" },
              { title: "At-Tawbah", subtitle: "(The Repentance)" },
              { title: "Yunus", subtitle: "(Jonah)" },
              { title: "Hud", subtitle: "(Hud)" },
              { title: "Yusuf", subtitle: "(Joseph)" }
            ].map((tile, index) => (
              <div 
                key={index}
                className="tile"
                onMouseEnter={() => handleTileHover(tile.title, true)}
                onMouseLeave={() => handleTileHover(tile.title, false)}
                style={{
                  borderRadius: isVerySmallMobile ? '8px' : isMobile ? '10px' : '12px',
                  borderWidth: isTablet ? '1.25px' : '1.5px'
                }}
              >
                <h4 style={{ fontSize: isVerySmallMobile ? '12px' : isMobile ? '14px' : '16px' }}>{tile.title}</h4>
                <p style={{ fontSize: isVerySmallMobile ? '10px' : isMobile ? '12px' : '14px' }}>{tile.subtitle}</p>
              </div>
            ))}
          </div>
        </div>
      </div>
      
      <Footer />
    </div>
  );
};

export default HomePage;